src/Engines/webgpuEngine.ts

```diff
@@ -344,50 +345,37 @@ export class WebGPUEngine extends Engine {
     /** @hidden */
     public dbgShowEmptyEnableEffectCalls = true;
 
-    private _snapshotRenderingRecordBundles = false;
-    private _snapshotRenderingPlayBundles = false;
-    private _snapshotRenderingMainPassBundleList: WebGPUBundleList[] = [];
-    private _snapshotRenderingModeSaved: number;
+    private _snapshotRendering: WebGPUSnapshotRendering;
 
     /**
      * Gets or sets the snapshot rendering mode
      */
     public get snapshotRenderingMode(): number {
-        return this._snapshotRenderingMode;
+        return this._snapshotRendering.mode;
     }
 
     public set snapshotRenderingMode(mode: number) {
-        if (this._snapshotRenderingRecordBundles) {
-            this._snapshotRenderingModeSaved = mode;
-        } else {
-            this._snapshotRenderingMode = mode;
-        }
+        this._snapshotRendering.mode = mode;
     }
 
     /**
      * Creates a new snapshot at the next frame using the current snapshotRenderingMode
      */
     public snapshotRenderingReset(): void {
-        this.snapshotRendering = false;
-        this.snapshotRendering = true;
+        this._snapshotRendering.reset();
     }
 
     /**
      * Enables or disables the snapshot rendering mode
      * Note that the WebGL engine does not support snapshot rendering so setting the value won't have any effect for this engine
      */
     public get snapshotRendering(): boolean {
-        return this._snapshotRenderingEnabled;
+        return this._snapshotRendering.enabled;
     }
 
     public set snapshotRendering(activate) {
-        this._snapshotRenderingMainPassBundleList.length = 0;
-        this._snapshotRenderingRecordBundles = this._snapshotRenderingEnabled = activate;
-        this._snapshotRenderingPlayBundles = false;
-        if (activate) {
-            this._snapshotRenderingModeSaved = this._snapshotRenderingMode;
-            this._snapshotRenderingMode = Constants.SNAPSHOTRENDERING_STANDARD; // need to reset to standard for the recording pass to avoid some code being bypassed
-        }
+        this._snapshotRendering.enabled = activate;
+        this._snapshotRenderingEnabled = activate; // keep this property in sync
```

Comment:
Do we need to keep this one ?

Reply:
You're right, I have removed some unused code.